crates/router/src/scheduler/workflows/create_auto_refund.rs

```diff
@@ -0,0 +1,126 @@
+use api_models::{
+    enums::EventType,
+    refunds::{RefundRequest, RefundType},
+    webhooks::OutgoingWebhookContent,
+};
+use common_utils::ext_traits::ValueExt;
+use diesel_models::{
+    enums::{EventClass, EventObjectType},
+    refund,
+};
+
+use super::{AutoRefundWorkflow, ProcessTrackerWorkflow};
+use crate::{
+    core::{
+        errors::ApiErrorResponse,
+        refunds::{add_auto_refund_task_to_process_tracker, refund_create_core},
+        webhooks::create_event_and_trigger_appropriate_outgoing_webhook,
+    },
+    errors,
+    logger::error,
+    routes::AppState,
+    types::storage,
+};
+
+#[async_trait::async_trait]
+impl ProcessTrackerWorkflow for AutoRefundWorkflow {
+    async fn execute_workflow<'a>(
+        &'a self,
+        state: &'a AppState,
+        process: storage::ProcessTracker,
+    ) -> Result<(), errors::ProcessTrackerError> {
+        let db = &*state.store;
+        let tracking_data: refund::AutoRefundWorkflow = process
+            .tracking_data
+            .clone()
+            .parse_value("AutoRefundWorkflow")?;
+        let payment_intent = tracking_data.payment_intent;
+        let retry_count = tracking_data.retry_count;
+        let max_retries = tracking_data.max_retries;
+        if retry_count > max_retries {
+            return Err(errors::ProcessTrackerError::FlowExecutionError {
+                flow: "RefundCreate",
```

Comment:
You could rename this to `AutoRefund` since that is the name of the workflow.

Reply:
Yes. Sure.